src/replica/replica_stub.cpp

```diff
@@ -87,6 +88,11 @@ replica_stub::replica_stub(replica_state_subscriber subscriber /*= nullptr*/,
       _fs_manager(false),
       _bulk_load_downloading_count(0)
 {
+
+    // make disk_engine destructed after replica_stub,
+    // because replica_stub relies on the former to close files.
+    disk_engine::instance();
```

Comment:
The destruction order in c++ is the reverse of the initialization order

Init:
1. replica_stub
2. disk_engine

Destruct:
1. disk_engine
2. replica_stub

If you want replica_stub destructed before aio, you should place this line in replication_service_app